lib/get-all-user-ids-to_local-file.rb

```diff
@@ -0,0 +1,39 @@
+# Usage - call script with bundle exec ruby lib/get-all-user-account-to_local-file-ok.rb | tee /Users/davidpye/alphagov/zendesk-scripts/data/users-to-delete.jsonl
+# above file path used in deletion step.
+
+require 'zendesk_api'
+require 'rest-client'
+require 'json'
+
+@client = ZendeskAPI::Client.new do |config|
+  config.url = ENV['ZENDESK_URL']
+  config.username = ENV['ZENDESK_USER_EMAIL']
+  config.password = ENV['ZENDESK_USER_PASSWORD']
+  config.retry = true
+end
+
+day = Date.today.prev_day
+lastyear = day - 365
+puts "last year:#{lastyear}"
+
+# TBC NEED TO LIMIT THE DATE RANGE AS 8000+ PAGES CURRENTLY!!!
+
+search_results = @client.search(:query => "type:user role:end-user -name:Zendesk organization_id:none")
+
+# The Zendesk API has 100 items per page, so programatically
+# determine how many pages we have by rounding to the nearest 100.
+
+user_count =  search_results.count
+number_of_pages = (user_count.to_f / 100).ceil
+
+# For diagnosis
+
+puts "User Accounts: #{user_count}"
```

Comment:
We're mixing JSON output with this "diagnostics" on stdout...if the ambition is to redirect stdout of this to a json input file this won't work as expected.

Reply:
@blairboy362 so maybe the 'for diagnosis' is misleading. The screen output allows the user to see that the script is doing stuff by counting pages, showing progress. The output to file needs to be json. Knowing the number of items and the fact of 100/page (will be in README shortly) allows them to estimate completion time. But I understand your meaning so have removed the 2 x puts statements. I'm unhappy because I've lost the ability to predict the run time but I understand why I need to remove them.